staging/src/k8s.io/apimachinery/pkg/util/constraints/constraints.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package constraints
```

Comment:
I wasn't expecting to export this for consumption outside the sets package... can we keep these unexpected and package-local?

Reply:
+1

Reply:
I built these type constraints into the sets package and named this file `ordered.go`